buildbot_nix/buildbot_nix/__init__.py

```diff
@@ -1368,6 +1384,8 @@ def nix_skipped_build_config(
         collapseRequests=False,
         env={},
         factory=factory,
+        do_build_if=lambda build: do_register_gcroot_if(build, branch_config_dict)
```

Comment:
`do_build_if` hasn't been backported yet.

Reply:
I thought, I had this reverted, but it was only locally.

Reply:
https://github.com/nix-community/buildbot-nix/pull/414

Reply:
Seems https://github.com/nix-community/buildbot-nix/pull/388 was going to backport it.

https://github.com/buildbot/buildbot/commit/f08eeef96e15c686a4f6ad52368ad08246314751 applies cleanly on 4.2.1 but I'm not sure if it needs anything else?

Reply:
Yes. we can backport this.

Reply:
https://github.com/nix-community/buildbot-nix/pull/419